src/Filters/remez_fir.jl

```diff
@@ -0,0 +1,845 @@
+        if ((nut*err-comp) <= 0.0) @goto L240; end
+        comp = nut * err;
+        @goto L235;
+      @label L240
+        klow = iext[j];
+        iext[j] = l+1;
+        j += 1;
+        jchnge += 1;
+        @goto L200;
+
+      @label L250
+        l = iext[j]+1;
+        if (jchnge > 0) @goto L215; end
+
+      @label L255
+        l += 1; if (l >= kup) @goto L260; end
+        err = (freq_eval(l,nz,grid,x,y,ad)-des[l]) * wt[l];
+        if ((nut*err-comp) <= 0.0) @goto L255; end
+        comp = nut * err;
+
+        @goto L210;
+      @label L260
+        klow = iext[j]; j += 1
+        @goto L200;
+
+      @label L300
+        if (j > nzz) @goto L320; end
+        if (k1 > iext[1] ) k1 = iext[1]; end
+        if (knz < iext[nz]) knz = iext[nz]; end
+        nut1 = nut;
+        nut = -nu;
+        l = 0;
+        kup = k1;
+        comp = ynz*(1.00001);
+        luck = 1;
+      @label L310
+        l += 1; if (l >= kup) @goto L315; end
+        err = (freq_eval(l,nz,grid,x,y,ad)-des[l]) * wt[l];
+        if ((nut*err-comp) <= 0.0) @goto L310; end
+        comp =  nut * err;
+        j = nzz;
+        @goto L210;
+
+      @label L315
+        luck = 6;
+        @goto L325;
+
+      @label L320
+        if (luck > 9) @goto L350; end
+        if (comp > y1) y1 = comp; end
+        k1 = iext[nzz];
+      @label L325
+        l = ngrid+1;
+        klow = knz;
+        nut = -nut1;
+        comp = y1*(1.00001);
+      @label L330
+        l -= 1; if (l <= klow) @goto L340; end
+        err = (freq_eval(l,nz,grid,x,y,ad)-des[l]) * wt[l];
+        if ((nut*err-comp) <= 0.0) @goto L330; end
+        j = nzz;
+        comp =  nut * err;
+        luck = luck + 10;
+        @goto L235;
+      @label L340
+        if (luck == 6) @goto L370; end
+        for j = 1 : nfcns
+            iext[nzz-j] = iext[nz-j];
+        end
+        iext[1] = k1;
+        @goto L100;
+      @label L350
+        kn = iext[nzz];
+        for j = 1 : nfcns
+            iext[j] = iext[j+1];
+        end
+        iext[nz] = kn;
+
+        @goto L100;
+      @label L370
+        ;
+        
+        if (jchnge <= 0)  # we are done if none of the extremal indices changed
+            break
+        end
+    end  # while
+    
+    # 
+    #    CALCULATION OF THE COEFFICIENTS OF THE BEST APPROXIMATION
+    #    USING THE INVERSE DISCRETE FOURIER TRANSFORM
+    #
+
+    a = vec(zeros(Float64, dimsize))   # frequency response on evenly spaced grid
+    p = vec(zeros(Float64, dimsize))
+    q = vec(zeros(Float64, dimsize)) 
+    alpha = vec(zeros(Float64, dimsize))   # return vector
+    
+    PI = 3.14159265358979323846
+    TWOPI = (PI+PI)
+    nm1 = nfcns - 1;   # nm1 => "nfcns minus 1"
+    fsh = 1.0e-06;
+    gtemp = grid[1];   # grid[1] is temporarily used in freq_eval in this loop
+    x[nzz] = -2.0;
+    cn  = 2*nfcns - 1;
+    delf = 1.0/cn;
+    l = 1;
+    kkk = 0;
+
+    if (bands[1] == 0.0 && bands[end] == 0.5) kkk = 1; end
```

Comment:
`kkk` also looks booleanesque to me. Any idea what it means? Can we get a more descriptive name?